catalog/src/main/java/it/eng/catalog/service/CatalogService.java

```diff
@@ -1,21 +1,23 @@
 package it.eng.catalog.service;
 
-import java.util.List;
-import java.util.Optional;
-
-import org.springframework.stereotype.Service;
-
 import it.eng.catalog.exceptions.CatalogErrorException;
 import it.eng.catalog.model.Catalog;
 import it.eng.catalog.model.Dataset;
 import it.eng.catalog.repository.CatalogRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
 
 /**
  * The CatalogService class provides methods to interact with catalog data, including saving, retrieving, and deleting catalogs.
  */
 @Service
 public class CatalogService {
-
+    @Autowired
+    private MongoTemplate mongoTemplate;
```

Comment:
Required or leftover?
If required, we should use constructor injection

Reply:
Nice catch, it is leftover - deleted!